### PR TITLE
Azure load tests

### PR DESCRIPTION
**Pull request by somesylvie**
# Description

Adding support for load testing in Azure.

## Issue

#1122

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 1]
```diff
@@ -226,6 +227,12 @@ protected DomainResponse handleMessageRequest(
         boolean markMetadataAsFailed = false;
         String errorMessage = "";
 
+        if ("True".equals(request.getHeaders().get("load-test"))
```
Comment (basiliskus) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:230]: Would it be better to do something like `Boolean.parseBoolean(request.getHeaders().get("load-test"))`? This handles casing and null safety

Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:230]: I like that.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 2]
```diff
@@ -173,6 +174,12 @@ DomainResponse handleResults(DomainRequest request) {
     }
 
     DomainResponse handleMetadata(DomainRequest request) {
+        if (Boolean.parseBoolean(request.getHeaders().get("load-test"))
+                && ApplicationContext.isPropertyPresent("REPORT_STREAM_URL_PREFIX")) {
```
Comment (saquino0827) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:178]: Is Application Context is set when we build the app?

File: operations/locustfile.py [hunk 3]
```diff
@@ -131,22 +142,38 @@ def assert_stats(environment):
 def get_auth_request_body():
     # set up the sample request body for the auth endpoint
     # using a valid test token found in the mock_credentials directory
-    auth_scope = "report-stream"
-    with open("mock_credentials/report-stream-valid-token.jwt") as f:
-        auth_token = f.read()
+
+    # TODO - notes/clarification on 2 different creds, plus expiration date of jwt
+    # TODO - do we want to TF the tests? If yes which envs? In CDC envs, may need to adjust IP allow list on app. Also set as private endpoints in test config?
```
Comment (saquino0827) [inline operations/locustfile.py:147]: This is a good question. My assumption is that we don't want load test bombarding CDC environments unless they're okay with that. If they aren't, this could just be an internal running load test. However, I feel if we only run it in internal, it loses value since that would mean we're only running these until we pass the application off to the CDC.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 2 above)
Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:178]: I'm having a hard time understanding your question here.  Can you rephrase it?

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 4]
```diff
@@ -226,6 +233,12 @@ protected DomainResponse handleMessageRequest(
         boolean markMetadataAsFailed = false;
         String errorMessage = "";
 
+        if (Boolean.parseBoolean(request.getHeaders().get("load-test"))
```
Comment (somesylvie) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:236]: Since any new places that call RS will also need to copy this code, we should document that (probably both in a comment here and the other location we evaluate this, plus in docs)

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 2 above)
Comment (saquino0827) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:177]: No worries, I think I answered my own silly question. I forgot the that the whole point of env variables was to allow them to be set differently for each environment lol.

**Review by basiliskus — APPROVED** [review]
